worker-taskunstowing/src/main/java/com/github/jobservice/workers/taskunstowing/TaskUnstowingWorker.java

```diff
@@ -0,0 +1,195 @@
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class TaskUnstowingWorker implements DocumentWorker
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(TaskUnstowingWorker.class);
+    // FAIL_ON_UNKNOWN_PROPERTIES is set to false because of:
+    // UnrecognizedPropertyException: Unrecognized field \"jobId\" (class com.hpe.caf.api.worker.TrackingInfo), not marked as ignorable
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper().configure(FAIL_ON_UNKNOWN_PROPERTIES, false);
+    private final DatabaseClient databaseClient;
+
+    public TaskUnstowingWorker(final DatabaseClient databaseClient)
+    {
+        this.databaseClient = databaseClient;
+    }
+
+    @Override
+    public void checkHealth(final HealthMonitor healthMonitor)
+    {
+        try {
+            databaseClient.checkHealth();
+        } catch (final Exception exception) {
+            final String message = "Database is unhealthy";
+            LOGGER.error(message, exception);
+            healthMonitor.reportUnhealthy(message);
+        }
+    }
+
+    @Override
+    public void processDocument(final Document document) throws InterruptedException, DocumentWorkerTransientException
+    {
+        LOGGER.info("Received request to unstow task(s)");
+
+        final String partitionId = document.getCustomData("partitionId");
+        if (Strings.isNullOrEmpty(partitionId)) {
+            final TaskUnstowingWorkerFailure failure = new TaskUnstowingWorkerFailure(
+                TaskUnstowingWorkerFailure.PARTITION_ID_MISSING_FROM_CUSTOM_DATA_ID,
+                "Custom data should contain a non-empty 'partitionId' property");
+            processFailure(failure, document);
+            return;
+        }
+
+        final String jobId = document.getCustomData("jobId");
+        if (Strings.isNullOrEmpty(jobId)) {
+            final TaskUnstowingWorkerFailure failure = new TaskUnstowingWorkerFailure(
+                TaskUnstowingWorkerFailure.JOB_ID_MISSING_FROM_CUSTOM_DATA_ID,
+                "Custom data should contain a non-empty 'jobId' property");
+            processFailure(failure, document);
+            return;
+        }
+
+        final WorkerTaskData workerTaskData = document.getTask().getService(WorkerTaskData.class);
+        if (workerTaskData == null) {
+            final TaskUnstowingWorkerFailure failure = new TaskUnstowingWorkerFailure(
+                TaskUnstowingWorkerFailure.FAILED_TO_GET_WORKER_TASK_DATA_ID,
+                "Failed to get worker task data");
+            processFailure(failure, document);
+            return;
+        }
+
+        LOGGER.info("Querying for stowed tasks for partition id {} and job id {}", partitionId, jobId);
+        final List<StowedTaskRow> stowedTaskRows;
+        try {
+            stowedTaskRows = databaseClient.getStowedTasks(partitionId, jobId);
```

Comment:
We'll need to test this.  It is a little dangerous.  If the number of tasks stored for a job was very large then this could cause an out of memory.  We need to see what the numbers are like in reality to know if we should do something more about this.

Reply:
I've added batch functionality now so we're limiting the number of rows read into memory.